src/widget/msApplicationLauncher.js

```diff
@@ -233,64 +237,96 @@ var MsApplicationButtonContainer = GObject.registerClass(
                     return false;
                 }
             });
-            this.highlightButton(this.filteredAppButtonList[0]);
+            //this.highlightButton(this.filteredAppButtonList[0]);
         }
 
-        highlightNextButton() {
-            let currentIndex = this.filteredAppButtonList.indexOf(
+        // Get current focused button index, resets to 0 if value is invalid
+        getCurrentIndex() {
+            const index = this.filteredAppButtonList.indexOf(
                 this.currentButtonFocused
             );
-            if (currentIndex != this.filteredAppButtonList.length - 1) {
+            if (index < 0 || index > this.maxIndex) {
```

Comment:
If the number of icons in the app grid is shown and the current highlighted icon is not shown anymore, the position of the highlighted app is reset to the first icon.

Reply:
The current behavior is to focus the first applications when the search changes which is for me the desired behavior. Since the first one should be the most accurate of the search result

Reply:
The code I likes is used only the number of icons in the grid changed. Otherwise, no app is focused but if you type any arrow keys or start typing, the fist icon is highlighted... That is the default behavior in Gnomes app grid.

Reply:
> Otherwise, no app is focused but if you type any arrow keys or start typing, the fist icon is highlighted (i.e. the first icon is actually the active one, just not highlighted until you press any key)... That is the default behavior in Gnomes app grid.

Yeah that's is good ! :+1: 

My only concern is about the current focused button when typing changes. Currently I move to the third application in my list then I decide to type a letter the focused app would be reset to be the first one. With your changes it's seem that it's would still focus the third app. Since the context (List of the app has changed) and they are sorted by most used I think we should reset the focus to the first one since the user couldn't predict which app will be on the "third" app button.

I'm so bad to explain my thought ^^ and I hop it's not too much frustrating for you !

Reply:
Yeah, the focus is reset only if the currently focused icon isn't shown anymore or ESC is pressed. I don't think it is much of a trouble since there are so few icons current, and it is just a temporary improvement until scrolling is implemented.

Don't worry about the explanations, I know what you mean when English isn't your first language and you try to explain something.

I would recommend that a dev branch is created, so PR are sent to that branch, it can be tested and when you think it is OK then merge to master and release a new version, makes testing and debugging easier IMHO.

Reply:
@Ifom ok I just did this so we can merge this on dev then I can edit some adjustment and if you want you can build upon this ? That's the idea right?

Reply:
Yep, I just did: the PR will be merged to dev.